### PR TITLE
[USJANUS-192] Try to extract id-token from the request :params if Bearer token is absent

### PR DESCRIPTION
**Pull request by DanBurton**

### CODE REVIEW

**Comment by rplevy** [comment]
We should add unit tests

File: src/ring_gatekeeper/authenticators/auth0.clj [hunk 1]
```diff
@@ -33,10 +33,15 @@
...
 
-(defn- extract-token [auth-header]
+(defn- extract-token-from-auth-header [auth-header]
   (let [[_ id-token] (re-matches #"^Bearer (.*)$" auth-header)]
     id-token))
 
+(defn- extract-token [request]
+  (let [auth-header (get-in request [:headers "authorization"] "")]
+    (or (extract-token-from-auth-header auth-header)
+        (get-in request [:params "id-token"]))))
```
Comment (dbolson) [inline src/ring_gatekeeper/authenticators/auth0.clj]: Is `id-token` arbitrary? If so, is there any way to parameterize it?

Comment (DanBurton) [inline src/ring_gatekeeper/authenticators/auth0.clj]: It follows suit with other places in our codebase that refer to this value, but I think within the scope of "our stuff" it is arbitrary.

**Comment by rplevy** [comment]
I guess the plan is to bump the version when deploying, after the merge?

**Comment by rplevy** [comment]
This will also require a PR on auth-proxy to use the new version

File: src/ring_gatekeeper/authenticators/auth0.clj (hunk 1 above)
Comment (dbolson) [inline src/ring_gatekeeper/authenticators/auth0.clj]: Looks like it's an auth0 thing, so :+1:

**Comment by DanBurton** [comment]
Yeah maybe let's add the version bump to both PRs.

**Comment by rplevy** [comment]
@DanBurton I've been banging my head on this, figuring out how it could possibly not be working.  It gets the token and the token is correct. However the authentication with auth0 fails in the ```(if-let [claims (jwt/validate token ...```

I've checkout master on all three repos and re-ran, back and forth twice and authorizations do work on master, yet not with these changes.

**Comment by rplevy** [comment]
The JWT validation works on master given the exact same token, for the 3 repositories, but not with the modified code in the 3 repositories.

File: src/ring_gatekeeper/authenticators/auth0.clj (hunk 1 above)
Comment (DanBurton) [inline src/ring_gatekeeper/authenticators/auth0.clj]: Note from @aprobus: document the need for `request` to have `:params` (e.g. use `wrap-params`).

**Pull request merged**